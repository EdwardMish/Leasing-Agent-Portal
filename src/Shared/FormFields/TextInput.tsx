import * as React from 'react';

import { useControlledForm } from './useControlledForm';
import {
    useValidationMessaging,
    useValidationMessagingArgs,
    ValidationMessageHandler,
} from './ValidationMessages';

const styles = require('./form-fields.module.css');

interface TextInputProps {
    formFieldId: string;
    inputValidation: useValidationMessagingArgs;
    label: string;
    name: string;
    isRequired?: boolean;
    initialValue?: string;
}

export const TextInput: React.FC<TextInputProps> = ({
    formFieldId,
    isRequired = false,
    inputValidation,
    label,
    name,
    initialValue,
}) => {
    const [inputValue, handler] = useControlledForm(initialValue);
    const [validationState, validationHandler] = useValidationMessaging(inputValidation);

    return (
        <>
            <label className={styles.label} htmlFor={formFieldId}>
                {label}
                {' '}
                {isRequired && <span className="required">(required)</span>}
            </label>
            <input
                className={`${styles.input} ${styles.inputText}`}
                id={formFieldId}
                name={name}
                type="text"
                value={inputValue}
                onChange={handler}
                onBlur={validationHandler}
            />
            <ValidationMessageHandler currentState={validationState} />
        </>
    );
};
