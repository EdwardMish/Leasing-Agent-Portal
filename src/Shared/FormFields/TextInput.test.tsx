import * as React from 'react';
import { render } from '@testing-library/react';

import { TextInput } from './TextInput';

describe('Component: TextInput', () => {
    const baseProps = {
        formFieldId: 'text-input-test',
        inputValidation: {
            applicableStates: [],
        },
        label: 'text-input-labe',
        name: 'text-input-test',
    };

    it('renders', () => {
        const { asFragment } = render(<TextInput {...baseProps} />);

        expect(asFragment()).toMatchSnapshot();
    });
});
