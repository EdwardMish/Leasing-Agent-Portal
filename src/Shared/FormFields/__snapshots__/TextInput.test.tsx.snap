// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: TextInput renders 1`] = `
<DocumentFragment>
  <label
    for="text-input-test"
  >
    text-input-labe 
  </label>
  <input
    class="undefined undefined"
    id="text-input-test"
    name="text-input-test"
    type="text"
    value=""
  />
</DocumentFragment>
`;
